packages/connect/package.json

```diff
@@ -10,55 +10,81 @@
+    "build:esm": "tsc --project tsconfig.json --module ES2015 --verbatimModuleSyntax --outDir ./dist/esm --declaration --declarationDir ./dist/esm && echo >./dist/esm/package.json '{\"type\":\"module\", \"sideEffects\":false}'",
+    "jasmine": "jasmine --config=jasmine.json",
+    "lint:types": "attw --pack"
   },
   "main": "./dist/cjs/index.js",
-  "types": "./dist/types/index.d.ts",
+  "types": "./dist/esm/index.d.ts",
   "exports": {
     ".": {
-      "types": "./dist/types/index.d.ts",
-      "import": "./dist/esm/index.js",
-      "require": "./dist/cjs/index.js"
+      "import": {
+        "types": "./dist/esm/index.d.ts",
+        "default": "./dist/esm/index.js"
+      },
+      "require": {
+        "types": "./dist/cjs/index.d.ts",
+        "default": "./dist/cjs/index.js"
+      }
     },
     "./protocol": {
-      "types": "./dist/types/protocol/index.d.ts",
-      "import": "./dist/esm/protocol/index.js",
-      "require": "./dist/cjs/protocol/index.js"
+      "import": {
+        "types": "./dist/esm/protocol/index.d.ts",
+        "default": "./dist/esm/protocol/index.js"
+      },
+      "require": {
+        "types": "./dist/cjs/protocol/index.d.ts",
+        "default": "./dist/cjs/protocol/index.js"
+      }
     },
     "./protocol-connect": {
-      "types": "./dist/types/protocol-connect/index.d.ts",
-      "import": "./dist/esm/protocol-connect/index.js",
-      "require": "./dist/cjs/protocol-connect/index.js"
+      "import": {
+        "types": "./dist/esm/protocol-connect/index.d.ts",
+        "default": "./dist/esm/protocol-connect/index.js"
+      },
+      "require": {
+        "types": "./dist/cjs/protocol-connect/index.d.ts",
+        "default": "./dist/cjs/protocol-connect/index.js"
+      }
     },
     "./protocol-grpc": {
-      "types": "./dist/types/protocol-grpc/index.d.ts",
-      "import": "./dist/esm/protocol-grpc/index.js",
-      "require": "./dist/cjs/protocol-grpc/index.js"
+      "import": {
+        "types": "./dist/esm/protocol-grpc/index.d.ts",
+        "default": "./dist/esm/protocol-grpc/index.js"
+      },
+      "require": {
+        "types": "./dist/cjs/protocol-grpc/index.d.ts",
+        "default": "./dist/cjs/protocol-grpc/index.js"
+      }
     },
     "./protocol-grpc-web": {
-      "types": "./dist/types/protocol-grpc-web/index.d.ts",
-      "import": "./dist/esm/protocol-grpc-web/index.js",
-      "require": "./dist/cjs/protocol-grpc-web/index.js"
+      "import": {
+        "types": "./dist/esm/protocol-grpc-web/index.d.ts",
+        "default": "./dist/esm/protocol-grpc-web/index.js"
+      },
+      "require": {
+        "types": "./dist/cjs/protocol-grpc-web/index.d.ts",
+        "default": "./dist/cjs/protocol-grpc-web/index.js"
+      }
     }
   },
   "typesVersions": {
     "*": {
       "protocol": [
-        "./dist/types/protocol/index.d.ts"
+        "./dist/esm/protocol/index.d.ts"
```

Comment:
Likewise with above, `typesVersions` is a fallback for older versions of TypeScript that do not prioritize subpath exports. However, it's unclear if this should be ESM or CJS?

Reply:
Same as in https://github.com/connectrpc/connect-es/pull/838#discussion_r1338555849, I'm expecting this to be relevant with the Node10 version resolution only, and since Node10 did not support ESM, these should point to the CJS types.

Reply:
Got it. Updated.